.github/workflows/ecr-push.yml

```
@@ -0,0 +1,39 @@
+name: authorize to ecr and generate images
+
+on:
+  workflow_call:
+    inputs:
+      aws-region:
+        description: "AWS cluster region"
+        type: string
+      ecr-repository:
+        description: "AWS repository"
+        type: string
+      image-tag:
+        description: "Tag for image build"
+        type: string
+
+jobs:
+  aws-auth-and-push:
+    name: build and push image to ECR Repo
+    runs-on: ubuntu-latest
+
+    steps:
+      - name: Check out code
+        uses: actions/checkout@v3
+
+      - name: Configure AWS credentials
+        uses: aws-actions/configure-aws-credentials@v1
+        with:
+          aws-access-key-id: ${{ secrets.AWS_ACCESS_KEY_ID }}
+          aws-secret-access-key: ${{ secrets.AWS_SECRET_ACCESS_KEY }}
```

We can use the following with https://github.com/Fatsoma/v2-terraform/pull/792 to set the IAM role (will need an extra input for `environment`):

```suggestion
          role-to-assume: arn:aws:iam::arn:aws:iam::819738237059:role/github-actions-${{ inputs.environment }}-${{ inputs.aws-region }}-Fatsoma-${{ github.event.repository.name }}
```

---

NB: Updated above to use a shorter IAM role to assume.